### PR TITLE
Additional env override when changing temporary db through Airflow CLI

### PR DESCRIPTION
**Pull request by jacek-izykowski**
## Description

Override AIRFLOW__DATABASE__SQL_ALCHEMY_CONN together with AIRFLOW__CORE__SQL_ALCHEMY_CONN when referring to temporary-database through Airflow CLI. This change is needed to support Airflow 2.3.3+.

## Checklist
- [X] I have followed [Sample Guidelines from AUTHORING_GUIDE.MD](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md)
- [X] README is updated to include [all relevant information](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#readme-file)
- [ ] **Tests** pass:   `nox -s py-3.9` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] **Lint** pass:   `nox -s lint` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [X] Please **merge** this PR for me once it is approved.
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample

### CODE REVIEW

File: composer/tools/composer_db_transfer.py [hunk 1]
```diff
@@ -3524,7 +3532,7 @@ def _update_airflow_db_connection(self: typing.Any) -> None:
             "airflow connections add airflow_db "
             "--conn-type postgres "
             "--conn-login $SQL_USER "
-            "--conn-password $SQL_PASSWORD "
+            '--conn-password="$SQL_PASSWORD" '
```
Comment (dandhlee) [inline composer/tools/composer_db_transfer.py:3535]: Should this be with an equal sign (`=`)? the other similar entries above don't come with it.

**Review by dandhlee — APPROVED** [review]
LGTM for Python, one minor comment.

**Review by leahecole — APPROVED** [review]

File: composer/tools/composer_db_transfer.py (hunk 1 above)
Comment (jacek-izykowski) [reply composer/tools/composer_db_transfer.py:3535]: This is correct and needed as SQL_PASSWORD might have '-' as its first character. If that happens, Airflow CLI considers the password as a separate option and option parsing fails.

Comment (dandhlee) [reply composer/tools/composer_db_transfer.py:3535]: Got it. Thank you!

**Pull request merged**